### DISCUSSION TITLE
Iris Security Alarm Keypad Model IL021 Support

### DISCUSSION THREAD

**Discussion by jchurchward** [Koenkk/zigbee2mqtt#6794] [Devices]
Hi All,

I don't suppose anyone has been working on support for the Iris White Security Alarm Keypad Model IL021? It's exactly like [this](https://www.ebay.com.au/itm/372642419625) device. If someone is working on this please let me know if I can supply anything that helps in it's development.

**Comment by jchurchward** [comment]
Anyone else working on this? I'd love to have my Iris keypad working again it's been a while :(

**Reply by floydcohen** [reply]
Just got this device, Iris Keypad V3 IL02_1.  Using the dev branch, it seems to work mostly if you add the 1112-S model to the Centralite 3400 entry before pairing.

I.e., node_modules/zigbee-herdsman-converters/devices/centralite.js
From
        zigbeeModel: ['3400-D', '3400'],
To
        zigbeeModel: ['3400-D', '3400', '1112-S'],

Of note for this model, you won't get any codes unless you arm it first.  Example in linux/mosquitto_pub:
mosquitto_pub  -h mqttserver -t 'zigbee2mqtt/Iris Keypad/set' -m '{ "arm_mode": { "mode": "arm_all_zones", "transaction":23 } }'

Then punching in 4 digits and OFF on the keypad, you'll receive something like this in mqtt
zigbee2mqtt/Iris Keypad {"action":"disarm","action_code":"1234","action_transaction":21,"action_zone":255,"battery":100,"linkquality":52,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad {"action":"","battery":100,"linkquality":52,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad/action disarm

You can disarm it with
jmosquitto_pub  -h mqttserver -t 'zigbee2mqtt/Iris Keypad/set' -m '{ "arm_mode": { "mode": "disarm" } }'
Once disarmed -
If you press ON button, you'll receive:
zigbee2mqtt/Iris Keypad {"action":null,"battery":100,"linkquality":60,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad {"action":"arm_all_zones","action_code":"","action_transaction":32,"action_zone":255,"battery":100,"linkquality":57,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad {"action":"","battery":100,"linkquality":57,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad/action arm_all_zones

If you press PARTIAL button:
zigbee2mqtt/Iris Keypad {"action":null,"battery":100,"linkquality":60,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad {"action":"arm_day_zones","action_code":"","action_transaction":34,"action_zone":255,"battery":100,"linkquality":60,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad {"action":"","battery":100,"linkquality":60,"occupancy":true,"temperature":22.4,"voltage":5900}
zigbee2mqtt/Iris Keypad/action arm_day_zones

PANIC button doesn't work, there's no code to handle that yet.

**Reply by jimmymarbles** [reply]
Hey, I have the Iris 3405-L and I want to do the same thing.  Unfortunately I haven't figured out how to access the dev branch or modify my local centralite..js file if possible.  Would you point me into the right direction?  Thanks!

I am running Home Assistant - Supervisor on Debian 11.

**Reply by floydcohen** [reply]
If you don't have the file node_modules/zigbee-herdsman-converters/devices/centralite.js
You might need to install zigbee-herdsman-converters via
npm install zigbee-herdsman-converters

**Reply by jimmymarbles** [reply]
Thanks for responding.  I was able to hack a external converter together and got the server to accept it.  I spent the afternoon trying to get the keypad to work as a panel for Alarmo, but it doesn't send codes consistently.  Basically I can get it to occasionally disarm, but not arm.  I don't know if its compatibility with the 3400-D or programming error.  I made sure I was next to the parent zigbee device to rule out distance, but I saw no difference.

I broke down and bought a used 3400 from ebay, so I will give that a shot.  If that doesn't work, I will throw in the towel and order a tablet...

**Reply by floydcohen** [reply]
To make it work a little more consistent, I had to setup an automation to rearm automatically after every disarm via homeassistant Mqtt publish.  So everyone has to know to enter codes and OFF.

**Reply by jimmymarbles** [reply]
Ah, that makes sense.  I'll try that when the 3400 arrives next week.  

You know, I would have though a physical keypad would be more in demand but it just isn't.  There are only used options now as everyone has just moved to phones and tablets.  I wanted the alarm system to stay simple and familiar.  I didn't want to go the tablet route as that could crash with some random OTA system update.  

However it may be what I have to get use to with migrating to HA from Smartthings.

**Reply by legop3** [reply]
this is a couple of years old now, but has anyone gotten some sort of working module for the iris keypad? I just got a zigbee transciever and I would like to use this keypad i found at the thrift store, along with some other (already supported) iris devices.
